internal/service/ec2/ec2_fleet.go

```diff
@@ -746,10 +762,20 @@ func resourceFleetDelete(ctx context.Context, d *schema.ResourceData, meta inter
 	conn := meta.(*conns.AWSClient).EC2Conn()
 
 	log.Printf("[DEBUG] Deleting EC2 Fleet: %s", d.Id())
-	output, err := conn.DeleteFleetsWithContext(ctx, &ec2.DeleteFleetsInput{
-		FleetIds:           aws.StringSlice([]string{d.Id()}),
-		TerminateInstances: aws.Bool(d.Get("terminate_instances").(bool)),
-	})
+
+	fleetType := d.Get("type").(string)
+
+	input := &ec2.DeleteFleetsInput{
+		FleetIds: aws.StringSlice([]string{d.Id()}),
+	}
+
+	if fleetType == ec2.FleetTypeInstant {
+		d.Set("terminate_instances", true)
+	}
+
+	input.TerminateInstances = aws.Bool(d.Get("terminate_instances").(bool))
```

Comment:
Can you please explain why the previous version always set `terminate_instances` and now you are wrapping it in conditional?

Reply:
Would remove this and force this check at construct